jpegenc/subblocks/bytestuffer/bytestuffer.py

```diff
@@ -0,0 +1,179 @@
+"""
+This module is MyHDL implementation
+of Byte Stuffer used for JPEG Encoder
+"""
+
+from myhdl import always_seq, always_comb
+from myhdl import Signal, concat, intbv, block
+
+
+class BSInputDataStream(object):
+    """
+    Input interface for the Byte Stuffer
+
+    data_in    : Input data to Byte Stuffer
+    read       : read signal sent to input FIFO
+    fifo_empty : asserts if input FIFO is empty
+    """
+    def __init__(self, width_data):
+        self.data_in = Signal(intbv(0)[width_data:])
+        self.read = Signal(bool(0))
+        self.fifo_empty = Signal(bool(0))
+
+
+class BScntrl(object):
+    """
+    Control Interface for Byte Stuffer
+
+    sof   : start of frame
+    start : send input frame when start asserts
+    ready : ready to access next frame
+    """
+    def __init__(self):
+        self.start = Signal(bool(0))
+        self.ready = Signal(bool(0))
+        self.sof = Signal(bool(0))
+
+
+class BSOutputDataStream(object):
+    """
+    Output Interface for the Byte Stuffer
+
+    byte       : output byte from the Byte Stuffer
+    addr       : output address to the RAM
+    data_valid : asserts when output data is valid
+    """
+    def __init__(self, width_data, width_addr_out):
+        self.byte = Signal(intbv(0)[width_data:])
+        self.addr = Signal(intbv(0)[width_addr_out:])
+        self.data_valid = Signal(bool(0))
+
+
+@block
+def bytestuffer(
+        clock, reset, bs_in_stream, bs_out_stream, bs_cntrl, num_enc_bytes):
+    """
+    Byte stuffer checks for 0xFF byte and adds a 0xFF00 Byte
+
+    Constants:
+
+    width_addr_out : maximum adress width of the output RAM
+    width_out      : width of the data in the ouput RAM
+
+    I/O Ports :
```

Comment:
Use the Google style docstrings as described in the best practices (this will make documentation easier in the long run by leveraging autodoc).